### PR TITLE
Deployment dashboard updates

### PR DESCRIPTION
**Pull request by simoncochrane**
* Add authentication to http endpoints (authorization is still to come but a service account is required to access).
* Refactor the data represented by a resource, which is now more kcd custom resource oriented than being per workload.
* Tweaks to the dashboard HTML template.
* Ensure stats have the correct prefix.

### CODE REVIEW

File: gok8s/workload/provider.go [hunk 1]
```diff
@@ -170,42 +169,37 @@ func (k *Provider) UpdateRolloutStatus(kcdName string, version, status string, t
...
-
-	specs, err := k.Workloads(kcd)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-
-	for _, spec := range specs {
-		resources = append(resources, spec.AsResource(kcd))
+// CVResource returns a resource summary for the given kcd instance.
+func (k *Provider) CVResource(kcd *kcdv1.KCD) *Resource {
```
Comment (suneeta-mall) [inline gok8s/workload/provider.go]: Oh this is left out as `CV` .. may be rename it to AsResource ?

File: k8s/ktmpl/kcd.yaml [hunk 2]
```diff
@@ -170,7 +170,7 @@ objects:
   - kind: Service
     apiVersion: v1
     metadata:
-      name: kcdapp
+      name: kcd-svc
```
Comment (suneeta-mall) [inline k8s/ktmpl/kcd.yaml]: Whats the reason for this change? I sort of liked them kcdapp all over ...

File: gok8s/workload/provider.go (hunk 1 above)
Comment (simoncochrane) [reply gok8s/workload/provider.go]: This was refactored in a subsequent PR.

File: k8s/ktmpl/kcd.yaml (hunk 2 above)
Comment (simoncochrane) [reply k8s/ktmpl/kcd.yaml]: Seems to be the convention for naming services.

**Pull request merged**